src/NuGet.Clients/PackageManagement.VisualStudio/VSSettings.cs

```diff
@@ -13,7 +13,23 @@ namespace NuGet.PackageManagement.VisualStudio
     [PartCreationPolicy(CreationPolicy.Shared)]
     public class VSSettings : Configuration.ISettings
     {
-        private Configuration.ISettings SolutionSettings { get; set; }
+        // to initialize SolutionSettings first time outside MEF constructor
+        private Configuration.ISettings _solutionSettings;
+
+        private Configuration.ISettings SolutionSettings
```

Comment:
Another problem here is `ResetSolutionSettings` has to run on UI thread due to number of DTE calls. According to [Threading Rule No.1](https://github.com/Microsoft/vs-threading/blob/v15.0/doc/threading_rules.md#1-if-a-method-has-certain-thread-apartment-requirements-sta-or-mta-it-must-either) it must start with `ThreadHelper.ThrowIfNotOnUIThread()`. After your change it's not guaranteed the call happens on UI thread anymore.